app/server.js

```diff
@@ -54,3 +57,63 @@ controller.on(['ambient', 'direct_message', 'file_share'], (bot, message) => {
 controller.hears(['show'], ['direct_message'], (bot, message) => {
   console.log(channelMessages)
 })
+
+/************************* milestones *************************/
+
+var channels = ['C9G17060H', 'C9G073P2N']   // dummy channel list
+
+// start and end dates for term and week count
+var startDate = moment('03/28/2018', 'MM/DD/YYYY')
+var endDate = moment('05/30/2018', 'MM/DD/YYYY')
+var currentWeek = 0
+
+// schedule milestone message for Wednesday after the Lab meeting at 8pm
+var rule = new schedule.RecurrenceRule() 
+rule.dayOfWeek = 3
+rule.hour = 20
+rule.minute = 0
+rule.second = 0 
+
+// if the current date is within the start and end date range
+if (moment().isSameOrAfter(startDate) && moment().isSameOrBefore(endDate) && currentWeek <= 8) {
+  
+  // post milestones message in each channel in channels list if it's the corresponding time
+  var milestones = schedule.scheduleJob(rule, function(){
+    for (var i = 0; i < channels.length;  i++) {
```

Comment:
This is a C of iterating. In Javascript, try using `forEach` since it's faster and more secure.